services/src/main/java/org/apache/druid/cli/CliRouter.java

```diff
@@ -91,6 +92,8 @@ public CliRouter()
           JsonConfigProvider.bind(binder, "druid.router.avatica.balancer", AvaticaConnectionBalancer.class);
           JsonConfigProvider.bind(binder, "druid.router.managementProxy", ManagementProxyConfig.class);
 
+          binder.bind(QuerySegmentWalker.class).to(FakeQuerySegmentWalker.class).in(LazySingleton.class);
```

Comment:
I would also add a comment mentioning that Router is just used to route the query to appropriate broker, so it is bound to a Noop walker.